netCDF4/utils.py

```diff
@@ -267,22 +258,19 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
             # The ellipsis stands for the missing dimensions.
             newElem.extend((slice(None, None, None),) * (nDims - len(elem) + 1))
             hasEllipsis = True
-        # Replace boolean array with slice object if possible.
-        elif ea.dtype.kind == 'b':
-            if ea.any():
-                indices = np.flatnonzero(ea)
-                start = indices[0]
-                stop = indices[-1] + 1
-                if len(indices) >= 2:
-                    step = indices[1] - indices[0]
-                else:
-                    step = None
-                if np.array_equal(indices, np.arange(start, stop, step)):
-                    newElem.append(slice(start, stop, step))
-                else:
-                    newElem.append(ea)
+        # Replace sequence of indices with slice object if possible.
+        elif np.iterable(e) and len(e) > 1:
```

Comment:
I would consider adding another case for `len(e) == 1`, which I expect is actually somewhat common. We could replace that with `slice(e[0], e[0] + 1, 1)`.